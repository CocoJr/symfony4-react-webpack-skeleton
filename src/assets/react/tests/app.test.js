import { createWaitForElement } from 'enzyme-wait';
import React from 'react';
import { MemoryRouter } from 'react-router';
import App from '../Core/app';

it('renders "/" without crashing', () => {
    let mockApiResponse = {
        status: 200,
        tasks: [
            {
                id: 1,
                title: "TEST"
            }
        ]
    };
    fetch.mockResponse(JSON.stringify(mockApiResponse));

    const wrapper = mount(
        <MemoryRouter initialEntries={[ '/' ]}>
            <App />
        </MemoryRouter>
    );

    return createWaitForElement('ul')(wrapper)
        .then(component => {
            wrapper.update();

            expect(component)
                .toMatchSnapshot();

            expect(component.find('ul').children('h1').length)
                .toBe(1);
            expect(component.find('ul').children('li').length)
                .toBe(mockApiResponse.tasks.length);
        });
});
