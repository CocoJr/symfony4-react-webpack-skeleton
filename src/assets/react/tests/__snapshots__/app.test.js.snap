// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "/" without crashing 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "tei9ht",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tei9ht",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <HashRouter>
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <Switch>
              <Route
                component={[Function]}
                computedMatch={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                exact={true}
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                path="/"
              >
                <Connect(Index)
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <Index
                    dispatch={[Function]}
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    isFetching={false}
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    tasks={
                      Array [
                        Object {
                          "id": 1,
                          "title": "TEST",
                        },
                      ]
                    }
                  >
                    <ul>
                      <h1>
                        Title
                      </h1>
                      <li
                        className="list-item"
                        key="1"
                      >
                        <Link
                          replace={false}
                          to="/tasks/1"
                        >
                          <a
                            href="#/tasks/1"
                            onClick={[Function]}
                          >
                             by 
                            TEST
                          </a>
                        </Link>
                      </li>
                    </ul>
                  </Index>
                </Connect(Index)>
              </Route>
            </Switch>
          </Router>
        </HashRouter>
      </Provider>
    </App>
  </Router>
</MemoryRouter>
`;
